core/src/main/scala/com/mesosphere/usi/core/CachedPendingLaunch.scala

```diff
@@ -0,0 +1,26 @@
+package com.mesosphere.usi.core
+import com.mesosphere.usi.core.models.PodId
+
+case class CachedPendingLaunch(pendingLaunch: Set[PodId] = Set.empty) {
+  /**
+    * Maintains a cache of pods pending launch, so a full scan isn't required on every offer
+    *
+    * Must be called after podSpecs are updated, and response effects are applied.
+    *
+    * @param podIds
+    */
+  def update(frame: Frame, podIds: Set[PodId]): CachedPendingLaunch = {
+    var newPendingLaunch = pendingLaunch
```

Comment:
What do you think builder `val builder = pendingLaunch.genericBuilder[PodId]` instead? Maybe it's to early to optimize it.

Reply:
I prefer the immutable builder